pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -673,6 +714,9 @@ func (r *MachineSyncReconciler) shouldMirrorCAPIMachineToMAPIMachine(ctx context
 	logger.V(4).WithName("shouldMirrorCAPIMachineToMAPIMachine").
 		Info("Checking if Cluster API machine should be mirrored", "machine", machine.GetName())
 
+	// TODO: should we check if the CAPI Machine is being deleted and do not bother creating a MAPI Machine mirror?
```

Comment:
I think it's a good idea :)

Reply:
Yeah we can deal with this in the deletion logic efforts PR

Reply:
Do we need to remove the comment here then?

Reply:
Missed this, removed.